torch_geometric/contrib/utils/fiber.py

```diff
@@ -0,0 +1,145 @@
+from collections import namedtuple
+from itertools import product
+from typing import Dict
+
+import torch
+from torch import Tensor
+
+FiberEl = namedtuple("FiberEl", ["degree", "channels"])
+
+
+class Fiber(dict):
```

Comment:
Yes, it does come from the paper, there is some good explanation in this [blog post](https://fabianfuchsml.github.io/se3iterative/). I do think it would be possible to do this without the Fiber class, we would just need to keep track of the degrees of the different features in `data.x` in some other field of `data`